src/System.Text.Json/src/System/Text/Json/JsonParser.cs

```diff
@@ -0,0 +1,565 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Text.Json
+{
+    public struct JsonParser : IDisposable
```

Comment:
Could you add comments explaining what each of the types is for. I now forgot, is this the reader that keeps track of buffer locations?